users/rverst/rverst.c

```diff
@@ -0,0 +1,392 @@
+#include "rverst.h"
+#include "print.h"
+
+#ifdef UNICODE
+#    include "unicode.h"
+#endif
+
+userspace_config_t userspace_config;
+
+int get_mode(void) {
+    int m = 0;
+    if (userspace_config.mode_1) {
+        m += 1;
+    }
+    if (userspace_config.mode_2) {
+        m += 2;
+    }
+    if (userspace_config.mode_3) {
+        m += 4;
+    }
+
+    return m;
+}
+
+void set_mode(int mode, bool save) {
+    if (mode == get_mode()) {
+        return;
+    }
+    switch_mode(mode);
+
+    if (mode > 7) {
+        mode = 7;
+    }
+
+    if (mode >= 4) {
+        userspace_config.mode_3 = true;
+        mode -= 4;
+    } else {
+        userspace_config.mode_3 = false;
+    }
+
+    if (mode >= 2) {
+        userspace_config.mode_2 = true;
+        mode -= 2;
+    } else {
+        userspace_config.mode_2 = false;
+    }
+
+    if (mode > 0) {
+        userspace_config.mode_1 = true;
+    } else {
+        userspace_config.mode_1 = false;
+    }
+
+    if (save) {
+        eeconfig_update_user(userspace_config.raw);
+    }
+}
+
+void switch_mode(int mode) {
+#ifdef UNICODE
+    switch (mode) {
+        case MAC_UNI:
+            set_unicode_input_mode(UC_MAC);
+            break;
+        case WINDOWS_UNI:
+            set_unicode_input_mode(UC_WINC);
+            break;
+        case LINUX_UNI:
+            set_unicode_input_mode(UC_LNX);
+            break;
+    }
+#endif
+}
+
+bool is_unicode(int mode) { return (mode == MAC_UNI) || (mode == WINDOWS_UNI) || (mode == LINUX_UNI); }
+
+//**********************
+// keyboard_pre_init
+//**********************
+__attribute__((weak)) void keyboard_pre_init_keymap(void) {}
+
+void keyboard_pre_init_user(void) {
+    userspace_config.raw = eeconfig_read_user();
+    switch_mode(get_mode());
+    keyboard_pre_init_keymap();
+}
+
+//************************
+// keyboard_post_init
+//************************
+__attribute__((weak)) void keyboard_post_init_keymap(void) {}
+
+void keyboard_post_init_user(void) {
+    // debug_enable = true;
+    // debug_matrix=true;
+    // debug_keyboard = true;
+
+#ifdef RGBLIGHT_ENABLE
+
+#endif
+
+    keyboard_post_init_keymap();
+}
+
+//**********************
+// eeconfig_init
+//**********************
+
+__attribute__((weak)) void eeconfig_init_keymap(void) {}
+
+void eeconfig_init_user(void) {
+    userspace_config.raw = 0;
+    eeconfig_update_user(userspace_config.raw);
+    eeconfig_init_keymap();
+    keyboard_init();
+}
+
+//**********************
+// process_record
+//**********************
+__attribute__((weak)) bool process_record_keymap(uint16_t keycode, keyrecord_t *record) { return true; }
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (keycode < SAFE_RANGE) {
+        return true;
+    }
```

Comment:
I wouldn't recommend this. You can alter the behavior of existing keycodes, if you want/need. 
```suggestion
```